central/complianceoperator/v2/report/manager/reportgenerator/report_gen_impl.go

```diff
@@ -0,0 +1,238 @@
+	checkResults "github.com/stackrox/rox/central/complianceoperator/v2/checkresults/datastore"
+	"github.com/stackrox/rox/central/graphql/resolvers"
+	"github.com/stackrox/rox/central/graphql/resolvers/loaders"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/branding"
+	"github.com/stackrox/rox/pkg/csv"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/notifier"
+	"github.com/stackrox/rox/pkg/notifiers"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/templates"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	reportGenCtx = resolvers.SetAuthorizerOverride(loaders.WithLoaderContext(sac.WithAllAccess(context.Background())), allow.Anonymous())
+
+	csvHeader = []string{
+		"Control Reference",
+		"Check(CCR)",
+		"Profile(version)",
+		"Check Description",
+		"Cluster",
+		"Status",
+		"Remediation",
+	}
+)
+
+type formatBody struct {
+	BrandedPrefix string
+	Profile       string
+	Pass          int
+	Fail          int
+	Mixed         int
+	Cluster       int
+}
+
+type formatSubject struct {
+	BrandedPrefix string
+	ScanConfig    string
+	Profiles      string
+}
+
+type complianceReportGeneratorImpl struct {
+	checkResultsDS        checkResults.DataStore
+	notificationProcessor notifier.Processor
+}
+
+type resultRow struct {
+	ClusterName string
+	CheckName   string
+	Profile     string
+	ControlRef  string
+	Description string
+	Status      string
+	Remediation string
+}
+
+type resultEmail struct {
+	resultCSVs map[string][]*resultRow
+	totalPass  int
+	totalFail  int
+	totalMixed int
+	profiles   []string
+	clusters   int
+}
+
+func (rg *complianceReportGeneratorImpl) ProcessReportRequest(ctx context.Context, req *ComplianceReportRequest) error {
+	//query compliance data
+	// Add the scan config name as an exact match
+
+	data, err := rg.getDataforReport(req, ctx)
+	if err != nil {
+		return err
+	}
+
+	zipData, err := rg.Format(data.resultCSVs)
+	if err != nil {
+		return err
+	}
+	var profiles string
+	for index, profile := range req.profiles {
+		if index == len(req.profiles)-1 {
+			profiles += fmt.Sprintf("%s", profile)
+			break
+		}
+		profiles += fmt.Sprintf("%s,", profile)
+	}
+	formatEmailBody := &formatBody{
+		BrandedPrefix: branding.GetCombinedProductAndShortName(),
+		Profile:       profiles,
+		Pass:          data.totalPass,
+		Fail:          data.totalFail,
+		Mixed:         data.totalMixed,
+		Cluster:       len(req.clusterIDs),
+	}
+
+	formatEmailSub := &formatSubject{
+		BrandedPrefix: branding.GetCombinedProductAndShortName(),
+		ScanConfig:    req.scanConfigName,
+		Profiles:      profiles,
+	}
+
+	return rg.sendEmail(zipData, formatEmailBody, formatEmailSub, req.notifiers, ctx)
+}
+
+func (rg *complianceReportGeneratorImpl) getDataforReport(req *ComplianceReportRequest, ctx context.Context) (*resultEmail, error) {
```

Comment:
I think we should separate this even further, though that can wait for the follow on work.  1 func to get the overview stats for the body of the email and another to get the actual results of the checks.

Reply:
Yes will do that in following prs